ansible_base/lib/jwt_auth/common/auth.py

```diff
@@ -0,0 +1,197 @@
+import logging
+from urllib.parse import urljoin, urlparse
+
+import jwt
+import requests
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from rest_framework.authentication import BaseAuthentication
+from rest_framework.exceptions import AuthenticationFailed
+
+logger = logging.getLogger("ansible_base.lib.jwt_auth.common.auth")
+
+
+default_mapped_user_fields = [
+    "first_name",
+    "last_name",
+    "email",
+    "is_superuser",
+    "is_system_auditor",
+]
+
+
+class JWTCommonAuth:
+    def __init__(self, user_fields=default_mapped_user_fields) -> None:
+        self.mapped_user_fields = user_fields
+
+    def parse_jwt_token(self, request):
+        logger.debug("Starting JWT Authentication")
+        token = request.headers.get("X-DAB-JW-TOKEN", None)
+        if not token:
+            logger.info("X-DAB-JW-TOKEN header not set for JWT authentication")
+            return None, None
+        logger.debug(f"Received token: {token}")
```

Comment:
For log hygiene, eventually, I would like to either remove some of these logs, or add more contextual information, since there are many "token" objects this could refer to. Also, could this token be really long? Maybe we could combine this with the prior `logger.debug` because it's better to have all the context on the same line.

Reply:
I'm torn on this because there are likely many tokens but there should only be one given the context of:
```
logger = logging.getLogger("ansible_base.lib.jwt_auth.common.auth")
```

Reply:
Changed message slightly but resolving since it has its own logger.